open_spiel/python/algorithms/deep_cfr.py

```diff
@@ -290,9 +297,12 @@ def _traverse_game_tree(self, state, player):
         sampled_regret[action] = expected_payoff[action]
         for a_ in state.legal_actions():
           sampled_regret[action] -= strategy[a_] * expected_payoff[a_]
-        self._advantage_memories[player].add(
-            AdvantageMemory(state.information_state_tensor(), self._iteration,
-                            advantages, action))
+      sampled_regret_arr = [0] * self._num_actions
+      for action in sampled_regret:
+        sampled_regret_arr[action] = sampled_regret[action]
+      self._advantage_memories[player].add(
+          AdvantageMemory(state.information_state_tensor(), self._iteration,
+                          sampled_regret_arr, action))
```

Comment:
I am not so sure about this as it sets SR for illegal actions to 0.

Reply:
Setting SR for illegal actions to 0 should be fine, I believe.

Reply:
Thanks @EricSteinberger, that was my next question. Truly appreciate the quick responses!